content/hardware/10.mega/shields/giga-display-shield/tutorials/12.emwin-guide/content.md

```diff
@@ -0,0 +1,634 @@
+In the sketch include the library like this:
+
+```arduino
+#include "DIALOG.h" /* emWin library includes Arduino_H7_Video and Arduino_GigaDisplayTouch library */
+```
+
+## General Set Up
+
+In this section, we will go through the fundamental elements of an emWin sketch:
+- How to define & configure the display,
+- how to create a grid layout,
+- how to add an object to the grid,
+- how to update the display.
+
+### emWin Screen Configuration
+
+When creating elements, information about the screen and placement needs to be provided. Let's create a pointer variable that can be used whenever the screenspace needs to be used.
+
+### Full Example
+
+```arduino
+void setup() {
+  /* Init SEGGER emWin library. It also init display and touch controller */
+  GUI_Init();
+
+  LCD_ROTATE_SetSel(1); /* Set landscape mode */
+  WM_MULTIBUF_Enable(1); /* Enable multi buffering mode for Windows manager */
+
+  /* Create the main window. It will include all the sub-windows */
+  WM_CreateWindowAsChild(0, 0, LCD_GetXSize(), LCD_GetYSize(), WM_HBKWIN, WM_CF_SHOW, _cbWin, 0);
+}
+```
+
+### Window Manager 
+
+The examples in this tutorial will be using the window manager method. This makes it easier to manage the screen elements and widgets. However, it is also possible to create emWin sketches without the window manager, which will be shown in this section.
+
+Here is an example that will make the screens background color red and print the text "Hello world" on the screen. Lets first take a look at how this would look without the window manager:
+
+```arduino
+#include <DIALOG.h>
+
+void setup() {
+  GUI_Init();
+  GUI_MULTIBUF_Begin();
+  GUI_SetBkColor(GUI_GREEN);
+  GUI_Clear();
+  GUI_SetColor(GUI_RED);
+  GUI_DispString("Hello world");
+  GUI_MULTIBUF_End();
+}
+
+void loop() {
+  GUI_Exec();
+}
+```
+
+And to get the same result with a window manager, it would look like this:
+
+```arduino
+#include "DIALOG.h"
+
+static void _cbWin(WM_MESSAGE * pMsg) {
+  switch (pMsg->MsgId) {
+  case WM_PAINT:
+    GUI_SetBkColor(GUI_GREEN);
+    GUI_Clear();
+    GUI_SetColor(GUI_RED);
+    GUI_DispString("Hello world");
+    break;
+  default:
+    WM_DefaultProc(pMsg);
+    break;
+  }
+}
+
+void setup() {
+  GUI_Init();
+  WM_MULTIBUF_Enable(1);
+  WM_CreateWindowAsChild(0, 0, LCD_GetXSize(), LCD_GetYSize(), WM_HBKWIN, WM_CF_SHOW, _cbWin, 0);
+}
+
+void loop() {
+  GUI_Exec();
+}
+```
+
+Using the window manager method is more complex and require more lines of code, but this method helps with keeping your GUI structured as it gets more elements and widgets to display. For example with the window manager it is impossible to overwrite areas outside a windows designated area. For a simple application not using a window manager might be good enough. Keep this in mind when developing your own emWin solutions! For more information, navigate over to the [emWin on Arduino documentation.](https://wiki.segger.com/emWin_on_Arduino#:~:text=Without%20the%20Window%20Manager%3A)
```

Comment:
```suggestion
Using the window manager method is more complex and require more lines of code, but this method helps with keeping your GUI structured as it gets more elements and widgets to display. For example with the window manager it is impossible to overwrite areas outside a windows designated area. For a simple application not using a window manager might be good enough. Keep this in mind when developing your own emWin solutions! For more information, navigate over to the [emWin on Arduino documentation](https://wiki.segger.com/emWin_on_Arduino#:~:text=Without%20the%20Window%20Manager%3A).
```

Reply:
Remember not to link punctuation in this way, read more [here](https://arduino.atlassian.net/wiki/x/x4Ai4Q).